simulator/mips/mips_instr.cpp

```diff
@@ -548,6 +552,16 @@ static const Table<I> isaMapCOP1_w =
     {0x21, { "cvt.d.w", mips_cvt_d_w<I>, OUT_FPU, 0, 'N', Imm::NO, Src1::CP1_FS, Src2::ZERO, Dst::CP1_FD, MIPS_I_Instr} },
 };
 
+template<typename I>
+static const Table<I> isaMapCOP1I =
+{
+    // Branches
+    {0x0, { "bc1f",  mips_bc1f<I>, OUT_FPU, 0, 'I', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::ZERO, MIPS_I_Instr} },
+    {0x1, { "bc1t",  mips_bc1t<I>, OUT_FPU, 0, 'I', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::ZERO, MIPS_I_Instr} },
+    {0x2, { "bc1fl", mips_bc1t<I>, OUT_FPU, 0, 'I', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::ZERO, MIPS_I_Instr} },
```

Comment:
FP31 register should be a source for CP1 branches

Reply:
I'm a little bit confused. If I set `FP31` as a source register `bc1f` will be disassembled as `bc1f $f31, offset`, not `bc1f offset`. Same with `c` instructions. Is that ok?

Reply:
I think yes. It might be not matching `objdump` exactly, but there are no goal to do this (objdump may substitute label names instead of jump/branch offsets, for instance). Moreover, it is more clear notation for microarchitects who do not care about ISA a lot, but care what is dependent on what.

Reply:
Ok, got that.